addon-test-support/-private/filter-test-modules.js

```diff
@@ -0,0 +1,51 @@
+const MODULE_PATH_REGEXP = /^(!?)\/([\w\W]*)\/(i?$)/;
+
+/**
+ * Return the matched test.
+ * e.g. if an input is '!/weight/' it returns an array, ['!/weight/', '!', 'weight', ''];
+ *
+ * @param {*} modulePathFilter
+ */
+function getRegexFilter(modulePathFilter) {
+  return MODULE_PATH_REGEXP.exec( modulePathFilter );
+}
+
+/**
+ * Return a list of test modules that contain a given module path string.
+ *
+ * @param {*} modules
+ * @param {*} modulePathFilter
+ */
+function stringFilter(modules, modulePathFilter) {
+  return modules.filter( module => module.includes(modulePathFilter) );
```

Comment:
Why do we need a separate stringFilter? Can regexFilter not handle this as well?